common/utils/src/main/scala/org/apache/spark/util/MavenUtils.scala

```diff
@@ -343,11 +343,31 @@ private[spark] object MavenUtils extends Logging {
 
   /* Set ivy settings for location of cache, if option is supplied */
   private[util] def processIvyPathArg(ivySettings: IvySettings, ivyPath: Option[String]): Unit = {
-    val alternateIvyDir = ivyPath.filterNot(_.trim.isEmpty).getOrElse {
+    val alternateIvyDir = ivyPath.filterNot(_.trim.isEmpty).map { path =>
+      // Expand tilde (~) in paths
+      if (path.startsWith("~")) {
+        val subPath = path.drop(1)
+        // Check for ~/foo or ~\foo or just ~
+        if (subPath.isEmpty || subPath.startsWith("/") || subPath.startsWith(File.separator)) {
```

Comment:
Does `File.separator` return `\` in a Windows environment? Is it necessary to check for both `\` and `File.separator`?